packages/@aws-cdk/aws-appconfig-alpha/lib/application.ts

```diff
@@ -0,0 +1,396 @@
+    }
+
+    return new Import(scope, id);
+  }
+
+  /**
+   * Imports an AWS AppConfig application into the CDK using its ID.
+   *
+   * @param scope The parent construct
+   * @param id The name of the application construct
+   * @param applicationId The ID of the application
+   */
+  public static fromApplicationId(scope: Construct, id: string, applicationId: string): IApplication {
+    const stack = cdk.Stack.of(scope);
+    const applicationArn = stack.formatArn({
+      service: 'appconfig',
+      resource: 'application',
+      resourceName: applicationId,
+    });
+
+    class Import extends ApplicationBase {
+      public readonly applicationId = applicationId;
+      public readonly applicationArn = applicationArn;
+      protected readonly extensible = new ExtensibleBase(scope, this.applicationArn);
+    }
+
+    return new Import(scope, id);
+  }
+
+  /**
+   * Retrieves the Lambda layer version Amazon Resource Name (ARN) for the AWS AppConfig Lambda extension.
+   *
+   * @param region The region for the Lambda layer (for example, 'us-east-1')
+   * @param platform The platform for the Lambda layer (default is Platform.X86_64)
+   * @returns Lambda layer version ARN
+   */
+  public static getLambdaLayerVersionArn(region: string, platform?: Platform): string {
+    return lambdaLayerVersions[platform || Platform.X86_64][region];
+  }
+
+  /**
+   * Adds the AWS AppConfig Agent as a container to the provided ECS task definition.
+   *
+   * @param taskDef The ECS task definition
+   */
+  public static addAgentToEcs(taskDef: ecs.TaskDefinition) {
+    taskDef.addContainer('AppConfigAgentContainer', {
+      image: ecs.ContainerImage.fromRegistry('public.ecr.aws/aws-appconfig/aws-appconfig-agent:latest'),
+      containerName: 'AppConfigAgentContainer',
+    });
+  }
+
+  /**
+   * The description of the application.
+   */
+  public readonly description?: string;
+
+  /**
+   * The name of the application.
+   */
+  public readonly name?: string;
+
+  /**
+   * The ID of the application.
+   */
+  public readonly applicationId: string;
+
+  /**
+   * The Amazon Resource Name (ARN) of the application.
+   */
+  public readonly applicationArn: string;
+
+  private _application: CfnApplication;
+  protected extensible: ExtensibleBase;
+
+  constructor(scope: Construct, id: string, props: ApplicationProps = {}) {
+    super(scope, id);
+
+    this.description = props.description;
+    this.name = props.name || Names.uniqueResourceName(this, {
+      maxLength: 64,
+      separator: '-',
+    });
+
+    this._application = new CfnApplication(this, 'Resource', {
+      name: this.name,
+      description: this.description,
+    });
+    this.applicationId = this._application.ref;
+    this.applicationArn = cdk.Stack.of(this).formatArn({
+      service: 'appconfig',
+      resource: 'application',
+      resourceName: this.applicationId,
+    });
+
+    this.extensible = new ExtensibleBase(scope, this.applicationArn);
```

Comment:
Where do you recommend documenting this?

Reply:
It depends on the reason, I guess. Is it because the constructs already extend `cdk.Resource`? If so, you could explain in the docstring of `ExtensibleBase` that it's supposed to be used with resources, but there is no way to inherit from two classes (at least within the jsii constraints).

Conversely, have you looked into making `ExtensibleBase` an abstract class that extends `cdk.Resource`?